### PR TITLE
Docker Desktop: update README with useful notes

### PR DESCRIPTION
**Pull request by stmcginnis**
## What this PR does / why we need it

This adds content to the extension README to give some details on how to
build and install the extension.

## Describe testing done for PR

Captured necessary steps as I walked through building and installing the extension.

### CODE REVIEW

**Review by jpmcb — COMMENTED** [review]
Marking myself for review - going to go through these steps this afternoon. Thanks Sean! This is amazing!!

File: extensions/docker-desktop/README.md [hunk 1]
```diff
@@ -1 +1,65 @@
...
+If you are making local changes and would like to try them out, you will need
+to follow these steps:
+
+1. In Docker Desktop, go to Preferences > Extensions and make sure
+   "Enable Docker Extensions" is checked.
+1. From a terminal, navigate to `$TCE_REPO/extensions/docker-desktop`.
+1. Run the following commands to build and install the local extension:
+
+   ```sh
+   make build-deps
```
Comment (jorgemoralespou) [inline extensions/docker-desktop/README.md:42]: These 2 steps can be simplified to only `make extension`

Comment (stmcginnis) [reply extensions/docker-desktop/README.md:42]: I got a bunch of failures when I tried to do that. Is there something else missing that needs to be done prior to this?

Comment (jorgemoralespou) [reply extensions/docker-desktop/README.md:42]: Maybe I need to revisit the makefile. Can you publish containers to ghcr.io for this project? Is this something desirable?

Comment (stmcginnis) [reply extensions/docker-desktop/README.md:42]: I do not know. Maybe that's something @joshrosso can weigh in on?

Comment (jpmcb) [reply extensions/docker-desktop/README.md:42]: Weird - looks like the `extension` target does both these steps:
https://github.com/vmware-tanzu/community-edition/blob/8434221ace624fad6f01b15d8c79de4ae5ee7f37/extensions/docker-desktop/Makefile#L45-L46

Comment (stmcginnis) [reply extensions/docker-desktop/README.md:42]: Oooh... I was mixing up `make build-extension` and `make extension`. I will update.

We may want to consider renaming and cleaning up the targets to be be a little more clear. ;)

**Review by jpmcb — APPROVED** [review]
Sweet!! Went through steps and success 👍🏼

**Comment by jpmcb** [comment]
I actually have one failure:

First got this:
```
#10 [internal] load metadata for ghcr.io/vmware-tanzu/community-edition-extension-for-dockerdesktop-tanzu-cli:main
#10 sha256:fbf412156d30ffbf7d2a78e1211e53ea5411d469da7b1dea1f7bb6bd3495765c
#10 ERROR: failed to authorize: rpc error: code = Unknown desc = failed to fetch anonymous token: unexpected status: 403 Forbidden

#8 [internal] load metadata for ghcr.io/vmware-tanzu/community-edition-extension-for-dockerdesktop-client:main
#8 sha256:d5629578e8da6b12f3386f19a88c28ca22dc0672a0d68276d903215d7d2b9f94
#8 ERROR: failed to authorize: rpc error: code = Unknown desc = failed to fetch anonymous token: unexpected status: 403 Forbidden

#9 [internal] load metadata for ghcr.io/vmware-tanzu/community-edition-extension-for-dockerdesktop-apps-bin:main
#9 sha256:6322e4fc2bec732974144b689a6da9e4703472bd85ed0b5ef4a297df9189266d
#9 ERROR: failed to authorize: rpc error: code = Unknown desc = failed to fetch anonymous token: unexpected status: 403 Forbidden
```

Then did `docker login` to `ghcr.io` and now:
```
#10 [internal] load metadata for ghcr.io/vmware-tanzu/community-edition-extension-for-dockerdesktop-apps-bin:main
#10 sha256:6322e4fc2bec732974144b689a6da9e4703472bd85ed0b5ef4a297df9189266d
#10 ERROR: ghcr.io/vmware-tanzu/community-edition-extension-for-dockerdesktop-apps-bin:main: not found
```

What am I missing here?

**Comment by jpmcb** [comment]
Fixed my above issue in https://github.com/vmware-tanzu/community-edition/pull/4391

I think this PR is good to go since the docs are correct and my issue was tangential.

**Pull request merged**